skimage/transform/_orientation.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+
+
+def _normalize(x):
+    """Normalizes an array.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to normalize.
+
+    Returns
+    -------
+    u : array
+        Unitary array.
+
+    Examples
+    --------
+    >>> x = np.arange(5)
+    >>> uX = _normalize(x)
+    >>> np.isclose(np.linalg.norm(uX), 1)
+    True
+    """
+    u = np.asarray(x)
+
+    norm = np.linalg.norm(u)
+
+    return u / norm
+
+
+def _axis_0_rotation_matrix(unit_vector, indices=None):
+    """Generate a matrix that rotates a vector to coincide with the 0th (y-)
+       coordinate axis.
+
+    Parameters
+    ----------
+    unit_vector : (N, ) array
+        Unit vector.
+    indices : sequence of int, optional
+        Indices of the components of `axis` that should be transformed.
+        If `None`, defaults to all of the indices of `axis`.
+
+    Returns
+    -------
+    rotation_matrix : (N, N) array
+        Orthogonal projection matrix.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+    .. [2] https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions
+
+
+    Examples
+    --------
+    >>> rotation_matrix = _axis_0_rotation_matrix([0, 1, 0])
+    >>> rotation_matrix @ [0, 1, 0]
+    array([ 1.,  0.,  0.])
+    """
+    ndim = len(unit_vector)
+
+    if indices is None:
+        indices = range(ndim)
+
+    rotation_matrix = np.eye(ndim)
+
+    # loop to create stages of 2D rotations around fixed axes
+    # that are multiplied to form our nD matrix; see: [2]_
+    for step in np.round(2 ** np.arange(np.log2(ndim))).astype(int):
+        plane_rotation_matrix = np.eye(ndim)
+
+        for n in range(0, ndim - step, step * 2):
+            if n + step >= len(indices):
+                break
```

Comment:
These two lines seem unnecessary? The `range` expression should ensure that this never happens?

If that is not the case for non-default indices, then maybe the range expression should be changed to stop at `len(indices) - step`?

Reply:
> These two lines seem unnecessary? The `range` expression should ensure that this never happens?

Sometimes `len(indices) < ndim - 2` in non-default indices

> If that is not the case for non-default indices, then maybe the range expression should be changed to stop at len(indices) - step?

I'm not 100% sure. If you can convince me then I'll change it, but I don't want to risk breaking anything.